templates/project/.github/workflows/publish.yml

```diff
@@ -34,27 +38,31 @@ jobs:
     - name: Check release type
       id: check-tag
       run: |
-        if echo ${{ github.event.ref }} | grep -Eq '^refs/tags/v.*rc[0-9]{1}$'; then
+        if echo {% raw %} ${{ github.event.ref }} {% endraw %} | grep -Eq '^refs/tags/v.*rc[0-9]{1}$'; then
```

Comment:
The syntax for liquid and github actions conflicts: https://github.com/cargo-generate/cargo-generate/discussions/843. So, any github thing in `${{ ... }` needs to be escaped.

The other way to do this would be to make this file explicitly not part of liquid substitution, and dig the project name out with `cargo metadata` and `jq`:

```
$ cargo metadata --no-deps --format-version=1 | jq '.packages[0].name' -r
awesome
```

We could put the result of that in `GITHUB_ENV` and then use it later.

I didn't do this because the docker image doesn't have `jq` in it, but we could add it of course. LMK if there is preference for that over this admittedly horrible escaping which will plague us indefinitely as we iterate the template workflow.

Reply:
why do you think this will hurt us in the long run?

Reply:
It will be easy to forget to wrap all github expansions in the `{% raw %}` and `{% endraw %}` tags, and if one is forgotten, it appears that `cargo generate` will just blithely continue on, producing an invalid workflow file, that won't be discovered until a tag is created.

Reply:
Can we isolate this issue by declaring this a variable for the workforce or something? Does such a thing exist?

Reply:
I've decided to go the `jq` path instead, and explicitly exclude the github workflow from liquid interpolation.